### DISCUSSION TITLE
Appsource App deployment to SAAS

### DISCUSSION THREAD

**Discussion by CPretorius1986** [microsoft/AL-Go#329] [Q&A] [answered]
Evening all,

I may be missing something very obvious.

We have setup a app we have on BC marketplace into AL-Go for a modernization of our development process, we are getting some weird error in our first instances for QA deployment

"Deploying to BC-GITHUB failed.  The remote server returned an error: (500) Internal Server Error. Internal Server Error {   "error": {     "code": "Unknown",     "message": "A supported MIME type could not be found that matches the content type of the response. None of the supported type(s) 'Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaType, Microsoft.OData.ODataMediaTyp...' matches the content type 'application/octet-stream'."   } } (ms-correlation-x = c953021d-4ce0-4651-8865-c3c21d7ec729)"

Then our 3rd instance deployment is giving the below which is more helpful.

"An extension with same app Id 'XXXX' already exists in the AppSource marketplace with publisher 'XX'. Please use the extension marketplace to install the extension from AppSource or use a new app Id for your per-tenant extension."

What I can summate is that the app is trying to install but then causing an issue because it is already in marketplace, how would/can I change the deployment to get this into a QA environment? I do not wish to change the app-id as it will be counter intuitive to being able to "test" the app as an in place deployment to a QA environment.

Any help in advance would be appreciated.

PS. where can I find information on the config for deployment/delivery of the compiled app to Appsource?

**Comment by jonaswre** [comment]
Hi,

If I understand correctly. You want to publish apps made for AppSource into QA (an online sandbox)

In that case you need to use impersonation, so that your apps will be published as dev. 

This can be done using the following auth-context

  $authContext = New-BcAuthContext -includeDeviceLogin

For reference check navcontainerhelper/Auth/New-BcAuthContext

**Reply by JanLehmkuhlCTM** [reply]
Can you please tell me what properties belong in the AuthContext string when using User_impersonation.

**Comment by freddydk** [comment, marked as answer]
TenantID, RefreshToken and Scopes - you can generate it by using:
```
New-BcAuthContext -includeDeviceLogin | New-ALGoAuthContext | Set-Clipboard
```
With the latest BcContainerHelper installed.

**Reply by JanLehmkuhlCTM** [reply]
I'm happy that I can now deliver our AppSource app to an online environment via GitHub. Thank you!
